syntax = "proto3";

package silenteight.datasource.api.transaction;

import "google/api/annotations.proto";
import "silenteight/datasource/api/common/common.proto";

option java_package = "com.silenteight.datasource.api.transaction";
option java_outer_classname = "TransactionInputProto";
option java_multiple_files = true;

/**
  Intended to be used by Free Text Agent
 */
service TransactionInputService {
  rpc BatchGetMatchTransactionInputsV1 (silenteight.datasource.api.common.BatchGetMatchAgentInputsRequest) returns (stream BatchGetMatchTransactionInputsV1Response) {
    option (google.api.http) = {
      post: "/agent-inputs/transaction/versions/1/matches:batchGet",
      body: "*"
    };
  }
}

message BatchGetMatchTransactionInputsV1Response {
  repeated TransactionInput transaction_inputs = 1;
}

message TransactionInput {
  string match = 1;
  repeated TransactionFeatureInput transaction_feature_inputs = 2;
}

message TransactionFeatureInput {
  string feature = 1;

  repeated string transaction_messages = 2;
  repeated WatchlistName watchlist_names = 3;
  WatchlistType watchlist_type = 4;

  // List of strings that screening engine matched (used only for Organization)
  repeated string matching_texts = 5;

  enum WatchlistType {
    INDIVIDUAL = 0;
    ORGANIZATION = 1;
  }
}

message WatchlistName {
  string name = 1;
  string type = 2;
}
